shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -263,26 +341,153 @@ private static int clampIndexToEditable(int index, Editable editable) {
     return clamped;
   }
 
+  public int getOffsetBefore(int offset) {
```

Comment:
Can you move this into a new `FlutterTextUtils.java` file?

I'd like to keep general purpose text utilities separate from the rest of `InputConnectionAdaptor` which handles interaction with Android's text input services.

Also add a comment explaining what `getOffsetBefore` does and why Android's similar APIs are insufficient.

Reply:
Done